modules/core/src/lib/loader-utils/parse-with-worker.js

```diff
@@ -9,7 +9,10 @@ export function canParseWithWorker(loader, data, options, context) {
     return false;
   }
   const loaderOptions = options && options[loader.id];
-  if (options.worker && loaderOptions && loaderOptions.workerUrl) {
+  if (
```

Comment:
Nit: This kind of complex expressions are  a little hard for me to read. 

Create a helper function `getWorkerUrl`, with separate if statements and comments?